pyg_lib/csrc/sampler/cpu/dist_neighbor_kernel.cpp

```diff
@@ -0,0 +1,261 @@
+        typedef std::pair<scalar_t, scalar_t> pair_scalar_t;
+        typedef std::conditional_t<!disjoint, scalar_t, pair_scalar_t> node_t;
+
+        const auto sampled_nodes_data =
+            sampled_nodes_with_dupl.data_ptr<scalar_t>();
+        const auto batch_data =
+            !disjoint ? nullptr : batch.value().data_ptr<scalar_t>();
+
+        std::vector<scalar_t> sampled_rows;
+        std::vector<scalar_t> sampled_cols;
+        auto mapper = Mapper<node_t, scalar_t>(num_nodes);
+
+        const auto seed_data = seed.data_ptr<scalar_t>();
+        if constexpr (!disjoint) {
+          mapper.fill(seed);
+        } else {
+          for (size_t i = 0; i < seed.numel(); ++i) {
+            mapper.insert({i, seed_data[i]});
+          }
+        }
+        size_t begin = 0, end = 0;
+        for (auto i = 0; i < sampled_nbrs_per_node.size(); i++) {
+          end += sampled_nbrs_per_node[i];
+
+          for (auto j = begin; j < end; j++) {
+            std::pair<scalar_t, bool> res;
+            if constexpr (!disjoint)
+              res = mapper.insert(sampled_nodes_data[j]);
+            else
+              res = mapper.insert({batch_data[j], sampled_nodes_data[j]});
+            sampled_rows.push_back(i);
+            sampled_cols.push_back(res.first);
+          }
+
+          begin = end;
+        }
+
+        std::tie(out_row, out_col) =
+            get_sampled_edges<scalar_t>(sampled_rows, sampled_cols, csc);
+      });
+
+  return std::make_tuple(out_row, out_col);
+}
+
+template <bool disjoint>
+std::tuple<c10::Dict<rel_type, at::Tensor>, c10::Dict<rel_type, at::Tensor>>
+relabel(const std::vector<node_type>& node_types,
+        const std::vector<edge_type>& edge_types,
+        const c10::Dict<node_type, at::Tensor>& seed_dict,
+        const c10::Dict<node_type, at::Tensor>& sampled_nodes_with_dupl_dict,
+        const c10::Dict<node_type, std::vector<int64_t>>&
+            sampled_nbrs_per_node_dict,
+        const c10::Dict<node_type, int64_t> num_nodes_dict,
+        const c10::optional<c10::Dict<node_type, at::Tensor>>& batch_dict,
+        const bool csc) {
+  if (disjoint) {
+    TORCH_CHECK(batch_dict.has_value(),
+                "Batch needs to be specified to create disjoint subgraphs");
+    for (const auto& kv : batch_dict.value()) {
+      const at::Tensor& batch = kv.value();
+      const at::Tensor& sampled_nodes_with_dupl = kv.value();
+      TORCH_CHECK(batch.is_contiguous(), "Non-contiguous 'batch'");
+      TORCH_CHECK(batch.numel() == sampled_nodes_with_dupl.numel(),
+                  "Each node must belong to a subgraph.'");
+    }
+  }
+  for (const auto& kv : seed_dict) {
+    const at::Tensor& seed = kv.value();
+    TORCH_CHECK(seed.is_contiguous(), "Non-contiguous 'seed'");
+  }
+  for (const auto& kv : sampled_nodes_with_dupl_dict) {
+    const at::Tensor& sampled_nodes_with_dupl = kv.value();
+    TORCH_CHECK(sampled_nodes_with_dupl.is_contiguous(),
```

Comment:
Same as above, can this check be moved to `dist_neighbor.cpp`?

Reply:
Moved.